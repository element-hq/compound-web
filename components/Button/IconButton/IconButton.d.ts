import React from "react";
import { UnstyledButtonPropsFor } from "../UnstyledButton";
type IconButtonProps = UnstyledButtonPropsFor<"button"> & JSX.IntrinsicElements["button"] & {
    /**
     * The CSS class name.
     */
    className?: string;
    /**
     * The avatar size in CSS units, e.g. `"24px"`.
     * @default 32px
     */
    size?: CSSStyleDeclaration["height"];
    /**
     * The icon button indicator displayed on the top right
     */
    indicator?: "default" | "highlight";
};
/**
 * Display an icon as a button. Can render an indicator
 */
export declare const IconButton: React.ForwardRefExoticComponent<Omit<React.PropsWithChildren<IconButtonProps>, "ref"> & React.RefAttributes<HTMLButtonElement>>;
export {};
//# sourceMappingURL=IconButton.d.ts.map