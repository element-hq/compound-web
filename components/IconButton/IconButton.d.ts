import React from "react";
type IconButtonProps = JSX.IntrinsicElements["button"] & {
    /**
     * The CSS class name.
     */
    className?: string;
    /**
     * The avatar size in CSS units, e.g. `"24px"`.
     * @default 32px
     */
    size?: CSSStyleDeclaration["height"];
    /**
     * The icon button indicator displayed on the top right
     */
    indicator?: "default" | "highlight";
};
/**
 * Display an icon as a button. Can render an indicator
 */
export declare const IconButton: React.ForwardRefExoticComponent<Omit<React.PropsWithChildren<IconButtonProps>, "ref"> & React.RefAttributes<HTMLButtonElement>>;
export {};
//# sourceMappingURL=IconButton.d.ts.map