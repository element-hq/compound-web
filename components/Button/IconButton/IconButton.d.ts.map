{"version":3,"file":"IconButton.d.ts","sourceRoot":"","sources":["../../../../src/components/Button/IconButton/IconButton.tsx"],"names":[],"mappings":"AAgBA,OAAO,KAAwC,MAAM,OAAO,CAAC;AAK7D,OAAO,EAAE,sBAAsB,EAAE,MAAM,mBAAmB,CAAC;AAE3D,KAAK,eAAe,GAAG,sBAAsB,CAAC,QAAQ,CAAC,GACrD,GAAG,CAAC,iBAAiB,CAAC,QAAQ,CAAC,GAAG;IAChC;;OAEG;IACH,SAAS,CAAC,EAAE,MAAM,CAAC;IACnB;;;OAGG;IACH,IAAI,CAAC,EAAE,mBAAmB,CAAC,QAAQ,CAAC,CAAC;IACrC;;OAEG;IACH,SAAS,CAAC,EAAE,SAAS,GAAG,WAAW,CAAC;CACrC,CAAC;AAEJ;;GAEG;AACH,eAAO,MAAM,UAAU,iIAyBrB,CAAC"}