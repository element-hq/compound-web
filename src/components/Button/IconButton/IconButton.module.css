/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.icon-button {
  --cpd-icon-button-indicator-border-size: calc(
    var(--cpd-icon-button-size) * 0.0625
  );

  inline-size: var(--cpd-icon-button-size);
  block-size: var(--cpd-icon-button-size);
  padding: calc(var(--cpd-icon-button-size) * 0.125);
  aspect-ratio: 1 / 1;
  color: var(--cpd-color-icon-tertiary);
  border: 0;
  appearance: none;
  cursor: pointer;
  border-radius: 50%;
  position: relative;
  background: transparent;
  line-height: 0;
}

.icon-button[aria-disabled="true"] {
  color: var(--cpd-color-icon-disabled);
  cursor: not-allowed;
}

.icon-button svg {
  inline-size: 100%;
  block-size: 100%;
}

.icon-button[data-indicator] svg {
  mask-image: url("./indicator-mask.svg");
  mask-position: center center;
  mask-repeat: no-repeat;
  mask-size: calc(var(--cpd-icon-button-size) * 0.75);
}

.icon-button[data-indicator]::before {
  content: "";
  position: absolute;
  inset-block-start: var(--cpd-icon-button-indicator-border-size);
  inset-inline-end: var(--cpd-icon-button-indicator-border-size);
  inline-size: calc(var(--cpd-icon-button-size) * 0.25);
  block-size: calc(var(--cpd-icon-button-size) * 0.25);
  border-radius: 50%;
  background: currentcolor;
}

.icon-button[data-indicator="highlight"]::before {
  background: var(--cpd-color-icon-success-primary);
}

/**
 * Hover state
 */

@media (hover) {
  .icon-button:not([aria-disabled="true"]):hover {
    color: var(--cpd-color-icon-primary);
    background: var(--cpd-color-bg-subtle-primary);
  }
}

.icon-button:not([aria-disabled="true"]):active {
  color: var(--cpd-color-icon-primary);
  background: var(--cpd-color-bg-subtle-primary);
}

@media (hover) {
  .icon-button:not([aria-disabled="true"])[data-indicator]:is(:hover)::before {
    /* Same colour as the background */
    border: var(--cpd-icon-button-indicator-border-size) solid
      var(--cpd-color-bg-subtle-primary);
    inset-block-start: 0;
    inset-inline-end: 0;
  }
}

.icon-button:not([aria-disabled="true"])[data-indicator]:is(:active)::before {
  /* Same colour as the background */
  border: var(--cpd-icon-button-indicator-border-size) solid
    var(--cpd-color-bg-subtle-primary);
  inset-block-start: 0;
  inset-inline-end: 0;
}
