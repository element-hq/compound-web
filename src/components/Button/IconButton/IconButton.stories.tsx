/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from "react";
import { Meta, StoryFn } from "@storybook/react";

import { IconButton as IconButtonComponent } from "./IconButton";

import UserIcon from "@vector-im/compound-design-tokens/icons/user-profile.svg";

export default {
  title: "Button/IconButton",
  component: IconButtonComponent,
  tags: ["autodocs"],
  parameters: {
    actions: { argTypesRegex: "^on.*" },
  },
  argTypes: {
    control: { type: "boolean" },
    onClick: { action: "onClick" },
  },
  args: {
    size: "32px",
    disabled: false,
    children: <UserIcon />,
  },
} as Meta<typeof IconButtonComponent>;

const Template: StoryFn<typeof IconButtonComponent> = (args) => (
  <>
    <IconButtonComponent {...args} size="32px">
      <UserIcon />
    </IconButtonComponent>

    <IconButtonComponent {...args} size="48px">
      <UserIcon />
    </IconButtonComponent>

    <IconButtonComponent {...args} size="64px">
      <UserIcon />
    </IconButtonComponent>
  </>
);

export const Demo = Template.bind({});

export const Default = { args: {} };
export const DefaultDisabled = {
  args: {
    disabled: true,
  },
};
export const WithIndicator = {
  args: {
    indicator: "default",
  },
};
export const WithIndicatorDisabled = {
  args: {
    indicator: "default",
    disabled: true,
  },
};
export const WithHighlightIndicator = {
  args: {
    indicator: "highlight",
  },
};
