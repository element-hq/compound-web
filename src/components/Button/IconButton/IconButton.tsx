/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { PropsWithChildren, forwardRef } from "react";
import classnames from "classnames";

import styles from "./IconButton.module.css";
import { UnstyledButton } from "../UnstyledButton";
import { UnstyledButtonPropsFor } from "../UnstyledButton";

type IconButtonProps = UnstyledButtonPropsFor<"button"> &
  JSX.IntrinsicElements["button"] & {
    /**
     * The CSS class name.
     */
    className?: string;
    /**
     * The avatar size in CSS units, e.g. `"24px"`.
     * @default 32px
     */
    size?: CSSStyleDeclaration["height"];
    /**
     * The icon button indicator displayed on the top right
     */
    indicator?: "default" | "highlight";
  };

/**
 * Display an icon as a button. Can render an indicator
 */
export const IconButton = forwardRef<
  HTMLButtonElement,
  PropsWithChildren<IconButtonProps>
>(function IconButton(
  { children, className, indicator, size = "32px", style, ...props },
  ref,
) {
  const classes = classnames(styles["icon-button"], className);
  return (
    <UnstyledButton
      as="button"
      ref={ref}
      className={classes}
      style={
        {
          "--cpd-icon-button-size": size,
          ...style,
        } as React.CSSProperties
      }
      {...props}
      data-indicator={indicator}
    >
      {React.Children.only(children)}
    </UnstyledButton>
  );
});
