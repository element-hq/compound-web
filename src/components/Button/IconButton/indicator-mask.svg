<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M24 7.6586C23.3745 7.8797 22.7012 8 22 8C18.6863 8 16 5.31372 16 2C16 1.29874 16.1201 0.625549 16.3413 0H0V24H24V7.6586Z" fill="black"/>
</svg>
