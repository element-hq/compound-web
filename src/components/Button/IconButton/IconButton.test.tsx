/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import React from "react";
import { composeStories } from "@storybook/react";

import * as stories from "./IconButton.stories";

const {
  Default,
  DefaultDisabled,
  WithIndicator,
  WithHighlightIndicator,
  WithIndicatorDisabled,
} = composeStories(stories);

describe("IconButton", () => {
  it("renders a Default IconButton", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
  it("renders a DefaultDisabled IconButton", () => {
    const { container } = render(<DefaultDisabled />);
    expect(container).toMatchSnapshot();
  });
  it("renders a WithIndicator IconButton", () => {
    const { container } = render(<WithIndicator />);
    expect(container).toMatchSnapshot();
  });
  it("renders a WithHighlightIndicator IconButton", () => {
    const { container } = render(<WithHighlightIndicator />);
    expect(container).toMatchSnapshot();
  });
  it("renders a WithIndicatorDisabled IconButton", () => {
    const { container } = render(<WithIndicatorDisabled />);
    expect(container).toMatchSnapshot();
  });
});
