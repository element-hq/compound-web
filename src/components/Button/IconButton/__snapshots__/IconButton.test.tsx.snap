// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconButton > renders a Default IconButton 1`] = `
<div>
  <button
    aria-disabled="false"
    class="_icon-button_e3253e"
    role="button"
    style="--cpd-icon-button-size: 32px;"
    tabindex="0"
  >
    <svg
      class="cpd-icon"
      fill="none"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.175 13.825C9.95833 14.6083 10.9 15 12 15C13.1 15 14.0417 14.6083 14.825 13.825C15.6083 13.0417 16 12.1 16 11C16 9.9 15.6083 8.95833 14.825 8.175C14.0417 7.39167 13.1 7 12 7C10.9 7 9.95833 7.39167 9.175 8.175C8.39167 8.95833 8 9.9 8 11C8 12.1 8.39167 13.0417 9.175 13.825ZM13.4125 12.4125C13.0208 12.8042 12.55 13 12 13C11.45 13 10.9792 12.8042 10.5875 12.4125C10.1958 12.0208 10 11.55 10 11C10 10.45 10.1958 9.97917 10.5875 9.5875C10.9792 9.19583 11.45 9 12 9C12.55 9 13.0208 9.19583 13.4125 9.5875C13.8042 9.97917 14 10.45 14 11C14 11.55 13.8042 12.0208 13.4125 12.4125Z"
        fill="currentColor"
      />
      <path
        d="M22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12ZM20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12Z"
        fill="currentColor"
      />
      <path
        d="M16.2297 18.7917C15.7472 18.6085 15.2623 18.4571 14.775 18.3375C13.8583 18.1125 12.9333 18 12 18C11.0667 18 10.1417 18.1125 9.225 18.3375C8.73774 18.4571 8.25283 18.6085 7.77028 18.7917C7.12696 18.3902 6.54483 17.8998 6.04104 17.3378C6.93176 16.9259 7.83474 16.6092 8.75 16.3875C9.81667 16.1292 10.9 16 12 16C13.1 16 14.1833 16.1292 15.25 16.3875C16.1653 16.6092 17.0682 16.9259 17.959 17.3378C17.4552 17.8998 16.873 18.3902 16.2297 18.7917Z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton > renders a DefaultDisabled IconButton 1`] = `
<div>
  <button
    aria-disabled="true"
    class="_icon-button_e3253e"
    role="button"
    style="--cpd-icon-button-size: 32px;"
    tabindex="0"
  >
    <svg
      class="cpd-icon"
      fill="none"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.175 13.825C9.95833 14.6083 10.9 15 12 15C13.1 15 14.0417 14.6083 14.825 13.825C15.6083 13.0417 16 12.1 16 11C16 9.9 15.6083 8.95833 14.825 8.175C14.0417 7.39167 13.1 7 12 7C10.9 7 9.95833 7.39167 9.175 8.175C8.39167 8.95833 8 9.9 8 11C8 12.1 8.39167 13.0417 9.175 13.825ZM13.4125 12.4125C13.0208 12.8042 12.55 13 12 13C11.45 13 10.9792 12.8042 10.5875 12.4125C10.1958 12.0208 10 11.55 10 11C10 10.45 10.1958 9.97917 10.5875 9.5875C10.9792 9.19583 11.45 9 12 9C12.55 9 13.0208 9.19583 13.4125 9.5875C13.8042 9.97917 14 10.45 14 11C14 11.55 13.8042 12.0208 13.4125 12.4125Z"
        fill="currentColor"
      />
      <path
        d="M22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12ZM20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12Z"
        fill="currentColor"
      />
      <path
        d="M16.2297 18.7917C15.7472 18.6085 15.2623 18.4571 14.775 18.3375C13.8583 18.1125 12.9333 18 12 18C11.0667 18 10.1417 18.1125 9.225 18.3375C8.73774 18.4571 8.25283 18.6085 7.77028 18.7917C7.12696 18.3902 6.54483 17.8998 6.04104 17.3378C6.93176 16.9259 7.83474 16.6092 8.75 16.3875C9.81667 16.1292 10.9 16 12 16C13.1 16 14.1833 16.1292 15.25 16.3875C16.1653 16.6092 17.0682 16.9259 17.959 17.3378C17.4552 17.8998 16.873 18.3902 16.2297 18.7917Z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton > renders a WithHighlightIndicator IconButton 1`] = `
<div>
  <button
    aria-disabled="false"
    class="_icon-button_e3253e"
    data-indicator="highlight"
    role="button"
    style="--cpd-icon-button-size: 32px;"
    tabindex="0"
  >
    <svg
      class="cpd-icon"
      fill="none"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.175 13.825C9.95833 14.6083 10.9 15 12 15C13.1 15 14.0417 14.6083 14.825 13.825C15.6083 13.0417 16 12.1 16 11C16 9.9 15.6083 8.95833 14.825 8.175C14.0417 7.39167 13.1 7 12 7C10.9 7 9.95833 7.39167 9.175 8.175C8.39167 8.95833 8 9.9 8 11C8 12.1 8.39167 13.0417 9.175 13.825ZM13.4125 12.4125C13.0208 12.8042 12.55 13 12 13C11.45 13 10.9792 12.8042 10.5875 12.4125C10.1958 12.0208 10 11.55 10 11C10 10.45 10.1958 9.97917 10.5875 9.5875C10.9792 9.19583 11.45 9 12 9C12.55 9 13.0208 9.19583 13.4125 9.5875C13.8042 9.97917 14 10.45 14 11C14 11.55 13.8042 12.0208 13.4125 12.4125Z"
        fill="currentColor"
      />
      <path
        d="M22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12ZM20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12Z"
        fill="currentColor"
      />
      <path
        d="M16.2297 18.7917C15.7472 18.6085 15.2623 18.4571 14.775 18.3375C13.8583 18.1125 12.9333 18 12 18C11.0667 18 10.1417 18.1125 9.225 18.3375C8.73774 18.4571 8.25283 18.6085 7.77028 18.7917C7.12696 18.3902 6.54483 17.8998 6.04104 17.3378C6.93176 16.9259 7.83474 16.6092 8.75 16.3875C9.81667 16.1292 10.9 16 12 16C13.1 16 14.1833 16.1292 15.25 16.3875C16.1653 16.6092 17.0682 16.9259 17.959 17.3378C17.4552 17.8998 16.873 18.3902 16.2297 18.7917Z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton > renders a WithIndicator IconButton 1`] = `
<div>
  <button
    aria-disabled="false"
    class="_icon-button_e3253e"
    data-indicator="default"
    role="button"
    style="--cpd-icon-button-size: 32px;"
    tabindex="0"
  >
    <svg
      class="cpd-icon"
      fill="none"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.175 13.825C9.95833 14.6083 10.9 15 12 15C13.1 15 14.0417 14.6083 14.825 13.825C15.6083 13.0417 16 12.1 16 11C16 9.9 15.6083 8.95833 14.825 8.175C14.0417 7.39167 13.1 7 12 7C10.9 7 9.95833 7.39167 9.175 8.175C8.39167 8.95833 8 9.9 8 11C8 12.1 8.39167 13.0417 9.175 13.825ZM13.4125 12.4125C13.0208 12.8042 12.55 13 12 13C11.45 13 10.9792 12.8042 10.5875 12.4125C10.1958 12.0208 10 11.55 10 11C10 10.45 10.1958 9.97917 10.5875 9.5875C10.9792 9.19583 11.45 9 12 9C12.55 9 13.0208 9.19583 13.4125 9.5875C13.8042 9.97917 14 10.45 14 11C14 11.55 13.8042 12.0208 13.4125 12.4125Z"
        fill="currentColor"
      />
      <path
        d="M22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12ZM20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12Z"
        fill="currentColor"
      />
      <path
        d="M16.2297 18.7917C15.7472 18.6085 15.2623 18.4571 14.775 18.3375C13.8583 18.1125 12.9333 18 12 18C11.0667 18 10.1417 18.1125 9.225 18.3375C8.73774 18.4571 8.25283 18.6085 7.77028 18.7917C7.12696 18.3902 6.54483 17.8998 6.04104 17.3378C6.93176 16.9259 7.83474 16.6092 8.75 16.3875C9.81667 16.1292 10.9 16 12 16C13.1 16 14.1833 16.1292 15.25 16.3875C16.1653 16.6092 17.0682 16.9259 17.959 17.3378C17.4552 17.8998 16.873 18.3902 16.2297 18.7917Z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton > renders a WithIndicatorDisabled IconButton 1`] = `
<div>
  <button
    aria-disabled="true"
    class="_icon-button_e3253e"
    data-indicator="default"
    role="button"
    style="--cpd-icon-button-size: 32px;"
    tabindex="0"
  >
    <svg
      class="cpd-icon"
      fill="none"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.175 13.825C9.95833 14.6083 10.9 15 12 15C13.1 15 14.0417 14.6083 14.825 13.825C15.6083 13.0417 16 12.1 16 11C16 9.9 15.6083 8.95833 14.825 8.175C14.0417 7.39167 13.1 7 12 7C10.9 7 9.95833 7.39167 9.175 8.175C8.39167 8.95833 8 9.9 8 11C8 12.1 8.39167 13.0417 9.175 13.825ZM13.4125 12.4125C13.0208 12.8042 12.55 13 12 13C11.45 13 10.9792 12.8042 10.5875 12.4125C10.1958 12.0208 10 11.55 10 11C10 10.45 10.1958 9.97917 10.5875 9.5875C10.9792 9.19583 11.45 9 12 9C12.55 9 13.0208 9.19583 13.4125 9.5875C13.8042 9.97917 14 10.45 14 11C14 11.55 13.8042 12.0208 13.4125 12.4125Z"
        fill="currentColor"
      />
      <path
        d="M22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12ZM20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12Z"
        fill="currentColor"
      />
      <path
        d="M16.2297 18.7917C15.7472 18.6085 15.2623 18.4571 14.775 18.3375C13.8583 18.1125 12.9333 18 12 18C11.0667 18 10.1417 18.1125 9.225 18.3375C8.73774 18.4571 8.25283 18.6085 7.77028 18.7917C7.12696 18.3902 6.54483 17.8998 6.04104 17.3378C6.93176 16.9259 7.83474 16.6092 8.75 16.3875C9.81667 16.1292 10.9 16 12 16C13.1 16 14.1833 16.1292 15.25 16.3875C16.1653 16.6092 17.0682 16.9259 17.959 17.3378C17.4552 17.8998 16.873 18.3902 16.2297 18.7917Z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;
